node/node.go

```diff
@@ -917,20 +919,22 @@ func defaultMetricsProvider(cfg *config.InstrumentationConfig) metricsProvider {
 		if cfg.Prometheus {
 			return &nodeMetrics{
 				consensus.PrometheusMetrics(cfg.Namespace, "chain_id", chainID),
-				p2p.PrometheusMetrics(cfg.Namespace, "chain_id", chainID),
+				indexer.PrometheusMetrics(cfg.Namespace, "chain_id", chainID),
 				mempool.PrometheusMetrics(cfg.Namespace, "chain_id", chainID),
+				p2p.PrometheusMetrics(cfg.Namespace, "chain_id", chainID),
+				proxy.PrometheusMetrics(cfg.Namespace, "chain_id", chainID),
 				sm.PrometheusMetrics(cfg.Namespace, "chain_id", chainID),
 				statesync.PrometheusMetrics(cfg.Namespace, "chain_id", chainID),
-				proxy.PrometheusMetrics(cfg.Namespace, "chain_id", chainID),
 			}
 		}
 		return &nodeMetrics{
 			consensus.NopMetrics(),
-			p2p.NopMetrics(),
+			indexer.NopMetrics(),
```

Comment:
this struct should definitely be keyed

Reply:
Oh my, I should absolutely have thought of that. Great call, thanks. Done.